### PR TITLE
Penalize Bad SSZ Payloads Correctly

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Networking Cleanup

**What does this PR do? Why is it needed?**

- [x] Penalize peer if they send bad payload over the wire.
- [x] Add unit test for it.

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: beacon-chain/sync/rpc.go [hunk 1]
```diff
@@ -115,14 +115,15 @@ func (s *Service) registerRPC(baseTopic string, handle rpcHandler) {
...
-				// Debug logs for goodbye/status errors
-				if strings.Contains(topic, p2p.RPCGoodByeTopic) || strings.Contains(topic, p2p.RPCStatusTopic) {
-					log.WithError(err).Debug("Could not decode goodbye stream message")
-					traceutil.AnnotateError(span, err)
-					return
-				}
 				log.WithError(err).Debug("Could not decode stream message")
 				traceutil.AnnotateError(span, err)
+				// In the event we get a goodbye which is unable to be unmarshalled, we set a backoff time.
+				if strings.Contains(topic, p2p.RPCGoodByeTopic) {
```
Comment (prestonvanloon) [inline beacon-chain/sync/rpc.go:121]: What about the status topic? That seems to be a logic change here.

**Review by prestonvanloon — CHANGES_REQUESTED** [review]
Discussed offline that we'd like to evaluate this a feature a bit more. There is a risk that non-ssz issues could end up banning peers unnecessarily. Given the time constraints for v1, we'll push this feature into v1.1 release target.

**Comment by prestonvanloon** [comment]
Any update on this?

**Comment by nisdas** [comment]
Will leave this on for v1.06 @prestonvanloon , I want to limit the amount of networking changes we push out to v1.05 for the week. A lot of big PRs are to be merged soon in v1.05 .

**Comment by rauljordan** [comment]
Should we still target this one for v1.0.6?

**Comment by nisdas** [comment]
@rauljordan Maybe 1.07 might be better for this, we have too many PRs now in 1.05 anyway. This also is a bit of a non trivial change to apply to the network might be better to have a bit more testing for it.

**Comment by prestonvanloon** [comment]
What can we do to merge this? Can we put it behind a feature flag?

Given that this PR has been open for 8 months, we should merge or close it this week. The stale bot will keep applying the label otherwise.

**Pull request closed without merging**